contracts/proxy-lib/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+use core::str;
+use std::collections::HashSet;
+
+use calimero_context_config::repr::Repr;
+use calimero_context_config::types::{ContextId, Signed, SignerId};
+use near_sdk::json_types::{Base64VecU8, U128, U64};
+use near_sdk::store::IterableMap;
+use near_sdk::{env, log, near, AccountId, Gas, PanicOnDefault, Promise, PromiseError};
+
+pub mod ext_config;
+pub use crate::ext_config::config_contract;
+
+pub type ProposalId = u32;
+
+#[derive(PartialEq, Debug)]
+#[near(serializers = [json, borsh])]
+pub struct ProposalWithApprovals {
+    pub proposal_id: ProposalId,
+    pub num_approvals: usize,
+}
+
+#[near(contract_state)]
+#[derive(PanicOnDefault)]
+pub struct ProxyContract {
+    pub context_id: Repr<ContextId>,
+    pub context_config_account_id: AccountId,
+    pub num_approvals: u32,
+    pub proposal_nonce: ProposalId,
+    pub proposals: IterableMap<ProposalId, ProposalWithSigner>,
+    pub approvals: IterableMap<ProposalId, HashSet<Repr<SignerId>>>,
+    pub num_proposals_pk: IterableMap<Repr<SignerId>, u32>,
+    pub active_proposals_limit: u32,
+}
+
+#[derive(Clone, Debug)]
+#[near(serializers = [borsh])]
+pub struct FunctionCallPermission {
+    allowance: Option<U128>,
+    receiver_id: AccountId,
+    method_names: Vec<String>,
+}
+
+// An internal request wrapped with the signer_pk and added timestamp to determine num_requests_pk and prevent against malicious key holder gas attacks
```

Comment:
Ideally we would turn this into Rustdoc documentation, i.e. using `///` and then following the standard formatting approach for function documentation 🙂

Reply:
Will do it in a following commit, leaving this open till that